src/cephadm/cephadm

```diff
@@ -3527,6 +3529,13 @@ class MgrListener(Thread):
 
     def handle_json_payload(self, data: Dict[Any, Any]) -> None:
         self.agent.ack = int(data['counter'])
+        if 'config' in data:
+            logger.info('Received new config from mgr')
+            config = data['config']
+            for filename in config:
+                with open(os.path.join(self.agent.daemon_dir, filename), 'w') as f:
+                    f.write(config[filename])
```

Comment:
What if `filename == "../../../../etc/shadow"`? I would only allow a fixed set of file names.

Reply:
now making sure only filenames in required_files are written out